test/test_integration.py

```diff
@@ -1078,8 +1126,6 @@ def socket_handler(listener):
             d.data = b'1234567890' * 2
             d.flags.add('END_STREAM')
             sock.send(d.serialize())
-
-            send_event.wait(5)
```

Comment:
We have to wait here when we close the socket. Another option is add `send_event.wait()` and `sock.close()` here, and `send_event.set()` in client side.